Source/WebCore/rendering/FloatingObjects.h

```diff
@@ -64,7 +64,7 @@ class FloatingObject {
     void setX(LayoutUnit x) { ASSERT(!isInPlacedTree()); m_frameRect.setX(x); }
     void setY(LayoutUnit y) { ASSERT(!isInPlacedTree()); m_frameRect.setY(y); }
     void setWidth(LayoutUnit width) { ASSERT(!isInPlacedTree()); m_frameRect.setWidth(width); }
-    void setHeight(LayoutUnit height) { ASSERT(!isInPlacedTree()); m_frameRect.setHeight(height); }
+    void setHeight(LayoutUnit height) { ASSERT(!isInPlacedTree() || isLowestPlacedFloatBottomInContainingBlock()); m_frameRect.setHeight(height); }
```

Comment:
I am not sure if "inContainingBlock" adds any value and it is rather misleading since you can have the lowest placed float in a containing block and yet not the lowest in the BFC which is what your trimming context is (actually this needs to be fixed in the function above) .

Reply:
FloatingObject seems to have multiple constructors and I am only seeing one initialization here.

Reply:
I changed it to "inBlockFormattingContext" to distinguish it from absolute coordinates, but let me know if you want me to change it. I also changed the implementation of the method to use the BFC root as well.